numpyro/infer/mcmc.py

```diff
@@ -72,7 +72,8 @@ def _get_num_steps(step_size, trajectory_length):
     return num_steps.astype(canonicalize_dtype(np.int64))
 
 
-def _sample_momentum(unpack_fn, mass_matrix_sqrt, rng_key):
+def momentum_generator(prototype_r, mass_matrix_sqrt, rng_key):
```

Comment:
cleanup: I want to move the logic to pack/unpack into a single function here. This way, we don't need to store a global attribute `momentum_generator` in `hmc`.